src/main/java/uk/gov/companieshouse/api/accounts/controller/CurrentPeriodController.java

```diff
@@ -100,6 +105,55 @@ public ResponseEntity create(@RequestBody @Valid CurrentPeriod currentPeriod,
         return responseEntity;
     }
 
+    @PutMapping
+    public ResponseEntity update(@RequestBody @Valid CurrentPeriod currentPeriod,
+        BindingResult bindingResult, @PathVariable("companyAccountId") String companyAccountId,
+        HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
+        if (smallFull.getLinks().get(SmallFullLinkType.CURRENT_PERIOD.getLink()) == null) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+        Errors errors = new Errors();
+
+        if (bindingResult.hasErrors()) {
+            errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult, errors);
```

Comment:
I thought we were going to return before doing the business rules type validation if binding result had errors? avoids us having to be extra careful with null checks etc in business logic

I'm ok with that changing but just double checking that this change is on purpose and will be the same across all resources as we add update methods to them

Reply:
Yes this statement is correct `I thought we were going to return before doing the business rules type validation if binding result had errors` the work for doing this is currently in another pull request, I have implemented this work in the old format. After the other PR is merged I will conduct a refactor to the correct format.